lua/hbac/command/subcommands.lua

```diff
@@ -0,0 +1,48 @@
+local autocommands = require("hbac.autocommands")
+local hbac_config = require("hbac.setup").opts
+local state = require("hbac.state")
+local utils = require("hbac.utils")
+
+local M = {}
+
+M.close_unpinned = function()
+	local buflist = utils.get_listed_buffers()
+	for _, bufnr in ipairs(buflist) do
+		if utils.buf_autoclosable(bufnr) then
+			hbac_config.close_command(bufnr)
+		end
+	end
+end
+
+M.toggle_pin = function()
+	local bufnr = vim.api.nvim_get_current_buf()
+	local pinned_state = state.toggle_pin(bufnr) and "pinned" or "unpinned"
+	return bufnr, pinned_state
+end
+
+M.toggle_all = function(pin_value)
+	local buflist = utils.get_listed_buffers()
+	for _, bufnr in ipairs(buflist) do
+		state.pinned_buffers[bufnr] = pin_value
+	end
+end
+
+M.pin_all = function()
+	M.toggle_all(true)
+end
+
+M.unpin_all = function()
+	M.toggle_all(false)
+end
```

Comment:
Is this ok regarding the `toggle_all` issue? The notifications are kept in the `command` module functions that call these functions so that the telescope picker can call these without spamming notifications. I don't know if keeping the name `toggle_all` makes sense, maybe `set_all` is more accurate now, but I didn't want to agonize over it.

Reply:
Set all is a good choice